src/streaming/zarr.common.cpp

```diff
@@ -0,0 +1,297 @@
+#include "macros.hh"
+#include "zarr.common.hh"
+
+#include <stdexcept>
+
+size_t
+zarr::bytes_of_type(ZarrDataType data_type)
+{
+    switch (data_type) {
+        case ZarrDataType_int8:
+        case ZarrDataType_uint8:
+            return 1;
+        case ZarrDataType_int16:
+        case ZarrDataType_uint16:
+            return 2;
+        case ZarrDataType_int32:
+        case ZarrDataType_uint32:
+        case ZarrDataType_float32:
+            return 4;
+        case ZarrDataType_int64:
+        case ZarrDataType_uint64:
+        case ZarrDataType_float64:
+            return 8;
+        default:
+            throw std::runtime_error("Invalid data type: " +
```

Comment:
How is this exception handled? Is it caught at a higher level and returned as an error code to the C interface?